sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -73,4 +74,14 @@ object DataWritingCommand {
       attr.withName(outputName)
     }
   }
+
+  def updateMetrics(
```

Comment:
`propogateMetrics` is probably more accurate.

Reply:
and let's add a comment to explain when this should be called.

Reply:
Done